src/components/Personalization/handlers/createProcessRedirect.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+export default ({ logger, executeRedirect, collect }) => item => {
+  const { content } = item.getData() || {};
+
+  if (!content) {
+    logger.warn("Invalid Redirect data", item.getData());
+    return {};
+  }
+
+  const render = () => {
+    return collect({ decisionsMeta: [item.getMeta()] }).then(() => {
+      executeRedirect(content);
```

Comment:
Render would be called only when `renderDecisions:true`, otherwise it should be just part of the array of propositions returned.
Collect should be called here with `documentMayUnload` argument.
You should try/catch errors here and log the error and showContainers.

Reply:
If renderDecisions is false, processRedirect isn't even called, and the proposition is just returned as part of the returned propositions. See https://github.com/adobe/alloy/blob/bf813fa5da1da3011d29ddcf357a3a55f7316878/src/components/Personalization/createFetchDataHandler.js#L63.

Changing it to use collect is fixed in #1061 

The try/catch is done in https://github.com/adobe/alloy/blob/bf813fa5da1da3011d29ddcf357a3a55f7316878/src/components/Personalization/handlers/createProcessPropositions.js#L23
The showContainers is in https://github.com/adobe/alloy/blob/bf813fa5da1da3011d29ddcf357a3a55f7316878/src/components/Personalization/createFetchDataHandler.js#L77